common/persistence/serialization/serializer.go

```diff
@@ -254,30 +259,68 @@ func (t *serializerImpl) serialize(p proto.Marshaler, encodingType enumspb.Encod
 }
 
 // NewUnknownEncodingTypeError returns a new instance of encoding type error
-func NewUnknownEncodingTypeError(encodingType enumspb.EncodingType) error {
-	return &UnknownEncodingTypeError{encodingType: encodingType}
+func NewUnknownEncodingTypeError(
+	encodingTypeStr string,
+	supportedEncoding ...enumspb.EncodingType,
```

Comment:
Why do we need this? Isn't it always just the valid enum values of enumspb.EncodingType?

Reply:
In some cases, we assume the encoding is always proto3 I think. e.g. https://github.com/temporalio/temporal/pull/3803/files#diff-4354d109cf3924c1736b9d080b790089ec74eaca2fdd174624c2a8bf02bdd587R187 

I should probably rename this to `expectedEncoding`